python/cucim/src/cucim/skimage/feature/corner.py

```diff
@@ -1370,7 +1370,7 @@ def corner_peaks(
     )
 
     if len(coords):
-        # TODO: modify to do KDTree on the GPU (cuSpatial?)
+        # TODO: modify to do KDTree on the GPU
         coords = cp.asnumpy(coords)
```

Comment:
Yeah as noted in the comment, this code falls back to the CPU for something that is best done on the GPU

At the time, we thought cuSpatial might provide a solution, in which case we could switch to cuSpatial for the GPU

However this comment (and the associated code) was written a while ago. As you note cuSpatial would not be the way to go today

That said, CuPy, with PR ( https://github.com/cupy/cupy/pull/7671 ), has added [`KDTree`]( https://docs.cupy.dev/en/latest/reference/generated/cupyx.scipy.spatial.KDTree.html ). This won't be available until CuPy 14.0.0

So instead would update this comment like so

```suggestion
        # TODO: modify to do KDTree on the GPU using CuPy
        # xref: https://github.com/rapidsai/cucim/issues/732
        coords = cp.asnumpy(coords)
```